File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let l=0, r=1
    let maxP = 0

    for(let i=0; i<prices.length; i++) {
        if(prices[r] - prices[l] > 0) {
            maxP = Math.max(maxP, prices[r] - prices[l])
            r++
        } else {
            l = r
            r = r + 1
        }
    }
    return maxP

};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l=0, r=1;
        int maxP = 0;
        while(r<prices.length) {
            if(prices[r] < prices[l]) {
                l = r;
                r = l+1;
            } else {
                maxP = Math.max(maxP, prices[r] - prices[l]);
                r++;
            }
        }
        return maxP;
    }
}